src/NonStrippingCreditCard.php

```diff
@@ -28,4 +28,102 @@ public function setNumber($value)
          */
         return $this->setParameter('number', $value);
     }
+
+    /**
+     * Sets the payment instrument name.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setPaymentInstrumentName($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('paymentInstrumentName', $value);
+    }
+
+    /**
+     * Gets the payment instrument name.
+     *
+     * @return string|null
+     */
+    public function getPaymentInstrumentName()
+    {
+        return $this->getParameter('paymentInstrumentName');
+    }
+
+    /**
+     * Sets the payment network.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setPaymentNetwork($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('paymentNetwork', $value);
+    }
+
+    /**
+     * Gets the payment network.
+     *
+     * @return string|null
+     */
+    public function getPaymentNetwork()
+    {
+        return $this->getParameter('paymentNetwork');
+    }
+
+    /**
+     * Sets the transaction identifier.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setTransactionIdentifier($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('transactionIdentifier', $value);
+    }
+
+    /**
+     * Gets the transaction identifier.
+     * 
+     * @return string|null
+     */
+    public function getTransactionIdentifier()
+    {
+        return $this->getParameter('transactionIdentifier');
+    }
+
+    /**
+     * Sets the token receieved from Apple Pay payment sheet.
+     * Includes the country, zip code, expiration date and account holder name.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setToken($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('paymentData', $value);
```

Comment:
this should be `token` to match the function name

Reply:
I thought that parameter needed to match the object's name?

Reply:
no, it actually needs to match the function name. the names of these parameters are only used by omnipay, they don't have anything to do with what comes back from vindicia, because we just take the stuff from vindicia and then pass it to this object, or pass it from this object to vindicia when building the request object.

when you're constructing a request for omnipay and pass it an array of string-indexed parameters, it actually uses the parameter name to identify which setters it should call for that object. so the parameter names actually have to match the setter names.